### PR TITLE
SetupCardTask Refactor and Stories Updated

### PR DESCRIPTION
**Pull request by chumpgrub**
- Updates SetupCardTask to be more flexible and intuitive.
- Updates SetupCard and SetupCardTask stories to reflect changes.

Here's a screenshot of the the diff when updating StoreBuilder to use the refactored component: http://p.tri.be/lOHm1J

### CODE REVIEW

File: src/components/setup-card-task/setup-card-task.stories.mdx [hunk 1]
```diff
@@ -109,87 +132,301 @@ export const Template = (args) => (
...
+that launch non-wizard functionality such as navigating the user to a 
+specific part of the admin area.
 
 ## Props
 <ArgsTable story="." />
 
-## Canvas
+## Usage
+- The `avatar` prop expects the <a href="https://mui.com/material-ui/api/avatar/" target="_blank">MUI Avatar</a> component, see the MUI documentation for details.
+  - If using an image for the avatar, we recommend a 128 x 128 PNG with transparent background, and the image/icon visually centered.
```
Comment (ColtenVanTussenbrook) [inline src/components/setup-card-task/setup-card-task.stories.mdx:146]: 👍

File: src/components/setup-card-task/setup-card-task.tsx [hunk 2]
```diff
@@ -156,21 +148,51 @@ const TaskActionArea = styled(CardActionArea)<TaskActionAreaProps>(({ theme }) =
...
+  const variantType = React.isValidElement(buttonProp) && variant === 'action' ? 'action' : 'task';
+
+  // If SetupCardTask is `disabled` add `disabled` prop to 
+  let button = (buttonProp && disabled)
+    ? React.cloneElement(buttonProp as React.ReactElement<any>, { disabled: true })
+    : buttonProp;
+
+  const { props: buttonProps } = button || {};
+
+  // If button has no `onClick` handler, we attribute the parent `onClick` handler.
```
Comment (ColtenVanTussenbrook) [inline src/components/setup-card-task/setup-card-task.tsx:187]: 😍

File: src/components/setup-card/setup-card.stories.mdx [hunk 3]
```diff
@@ -33,6 +33,8 @@ import * as SetupCardTaskStories from '../setup-card-task/setup-card-task.storie
 import * as SetupCardFooterStories from '../setup-card-footer/setup-card-footer.stories.mdx';
 import { theme } from '../../theme';
 
+console.log(SetupCardFooterStories);
```
Comment (ColtenVanTussenbrook) [inline src/components/setup-card/setup-card.stories.mdx:36]: did you want this here?

**Review by ColtenVanTussenbrook — APPROVED** [review]
Looks really good, I like the changes!

File: src/components/setup-card/setup-card.stories.mdx (hunk 3 above)
Comment (chumpgrub) [reply src/components/setup-card/setup-card.stories.mdx:36]: Dangit!

**Pull request merged**